### ISSUE TITLE
Message scope doesn't work with the 'page' display of Text element

### ISSUE THREAD

**Issue by innoeduke** [Chainlit/chainlit#15]
While text element in 'page' display was sent, message scope didn't work. Here're the repo steps:

Step 1: Run this sample code snippet
```
#code example
@cl.on_message
def main(message: str):
    elements = [
        cl.Text(message, name='side_text', display='side'),
        cl.Text(message, name='page_text', display='page'),
        cl.Text(message, name='inline_text', display='inline'),
    ]

    cl.send_message(content=f"Received: {message}", elements=elements)
```

Step 2: Send two messages
<img width="1058" alt="image" src="https://github.com/Chainlit/chainlit/assets/20149014/4539ecfb-36a0-4a99-a515-d7ef084d3bef">

Step 3: Click the page_text link in the 2nd message. 

Expected to get an output message: "2nd msg: side_text, page_text, inline_text", but still got the 1st message output. 

<img width="535" alt="image" src="https://github.com/Chainlit/chainlit/assets/20149014/62cbecf8-563d-4890-8b3c-c6b39f6b9c72">

**Comment by willydouhard** [comment]
Good catch, this is because on the page view we check for the element to display by name (as you can see in the URL) and in that case you have multiple elements with the same name.

While we fix this, you can avoid this behavior by specifying different names for page elements

**Comment by willydouhard** [comment]
this should be fixed with the 0.2.108 release
